### PR TITLE
Feature/content access control

### PR DESCRIPTION
**Pull request by markconroy**
1. Add the LGD access control module to the site
2. Configures the site to have a field called "Access Control" on all content types
3. Adds a "Devolved Editor" role to the site
4. Add an "Access Control" vocabulary that is used to power the "Site Sections" workbench access scheme
5. Allows "User Manager" role to add people to the site sections
6. Allows "Editor" role to create/edit/delete terms in the Access Control vocabulary

Note, since all content types are under access control, **NO EDITOR** will be able to edit content until they are added to a "Site Section" at `/admin/config/workflow/workbench_access`. The best thing to do will be to add each current editor to the full "Site Section" section at `/admin/config/workflow/workbench_access/site_section/sections` so they can continue to work as normal. Then, we can add some terms to the Access Control vocabulary and situate users within specific site sections.

### CODE REVIEW

**Review by siliconmeadow — CHANGES_REQUESTED** [review]
Hi @markconroy - just need to update your lock file with a trusty `ddev composer update --lock`

```
# Lock file errors
- The lock file is not up to date with the latest changes in composer.json, it is recommended that you run `composer update` or `composer update <package name>`.
```

You can always feel free to do a `ddev composer tests-required` which does `composer validate`, runs `phpcs` `phpunit` etc. any time, btw.

**Comment by Polynya** [comment]
Hi @markconroy, I'm looking forward to working with you! Can we discuss this with Will next week? We're using Group for another council so I'd like to compare the approaches. Also, there are a lot of config changes which will conflict with those in our release branch, and I'd like to get them into develop first.

File: config/default/core.entity_view_display.node.localgov_event.teaser.yml [hunk 1]
```diff
@@ -36,6 +37,11 @@ content:
     third_party_settings: {  }
     weight: 2
     region: content
+  content_moderation_control:
```
Comment (Polynya) [inline config/default/core.entity_view_display.node.localgov_event.teaser.yml:40]: Content moderation control should be hidden on all display modes.

**Comment by willguv** [comment]
Hi all, happy to discuss this here if you are?

**Comment by markconroy** [comment]
@siliconmeadow can you run that lock file update. I don't think the error in the lock file is from my side. My only change to the lock file is from this commit https://github.com/essexcountycouncil/essex-gov-uk-drupal/pull/336/commits/055764e86003ec5f879c413022a5d79f876aa956 and that only added a new module, didn't update anything else in the codebase. Also, I don't have access to the ci/gitlab/gitlab.nomensa.com to see what errors are being reported.

File: config/default/core.entity_view_display.node.localgov_event.teaser.yml (hunk 1 above)
Comment (markconroy) [reply config/default/core.entity_view_display.node.localgov_event.teaser.yml:40]: This should be fixed now.

**Comment by Polynya** [comment]
> I have the composer.lock file updated now, but `composer tests-required` is still reporting some issues, but they are not with any module(s) that I am working on so I'll leave you to check on those ones if that's okay.
> 
> ```
> - require.ckeditor/div : exact version constraints (4.10.1) should be avoided if the package follows semantic versioning
> - require.localgovdrupal/localgov_eu_cookie_compliance : exact version constraints (1.x-dev) should be avoided if the package follows semantic versioning
> - require.localgovdrupal/localgov_forms : exact version constraints (1.x-dev) should be avoided if the package follows semantic versioning
> - require.wa72/htmlpagedom : unbound version constraints (*) should be avoided```
> ```

Those warnings are ok to ignore. We'll resolve them separately

**Comment by markconroy** [comment]
> What does Essex need to achieve
An access control mechanism that is easy to implement and adminsiter and can also be contributed back to LGD so other councils can follow the same pattern and not reinvent the wheel.

> What might they need in the future
I'll have to leave this to @willguv to answer

> Why Workbench Access was chosen instead of Group, which is already implemented on Intranet for content moderation but is not actively used
Workbench access was chosen as it's a very popular module for this type of feature and was developed specifically to cater to this use case. 

Group was not chosen as it is a much larger module with a wider remit. Contributing a custom approach to the LGD community based on WA is a much easier feat than proposing that councils start adding Group module and all it's overheads to their CMS.

> Is it ok that the features on Gov and Intranet will be different, or would Group be removed from Intranet?
I'll have to leave this to @willguv to answer. That said, if group is implemented but not being used, I guess it wouldn't be a huge task to remove it and swap in the WA approach instead.

**Comment by willguv** [comment]
Hi Adrian, 

Thanks for sense checking this. There's a urgency our side to get this up and running. Could you give me an estimate for deploying the work please?

On your questions:

The high level need is for content designers to assign publishing permissions at a section level. I'm happy to go through the specifics when there's more time.

From conversations with content designers, it's very unlikely this need will change over time (eg publishing permissions in one section, review in another).

It's not ideal that we have potentially two different ways of assigning access rights. I'm planning to chat with Suchi about intranet's use of Group, and we'll rationalise this at a later date.

We skipped the initial solutions conversation here, which also isn't great. I'll make sure everyone is looped in earlier for future work.

**Comment by Polynya** [comment]
Thanks @willguv, @markconroy. Unfortunately, as this needs to be deployed urgently, it should be merged straight into main branch. Develop branch has significant changes to config and the theme which aren't ready to be deployed yet. Sorry, I guess it's not going to be simple to rebase with so many config changes.

**Comment by markconroy** [comment]
Closing this. Will open a PR against `main` instead.

**Pull request closed without merging**